app/ante/min_commission.go

```diff
@@ -0,0 +1,76 @@
+package ante
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+type MinCommissionDecorator struct {
+	codec codec.BinaryCodec
+}
+
+func NewMinCommissionDecorator(codec codec.BinaryCodec) MinCommissionDecorator {
+	return MinCommissionDecorator{
+		codec,
+	}
+}
+
+func checkCommission(m sdk.Msg) error {
+	switch msg := m.(type) {
+	case *stakingtypes.MsgCreateValidator:
+		c := msg.Commission
+		if c.Rate.LT(sdk.NewDecWithPrec(5, 2)) {
+			return sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "commission can not be lower than 5%")
+		}
+	case *stakingtypes.MsgEditValidator:
+		// if commission rate is nil, it means only other fields are being update and we must skip this validation
+		if msg.CommissionRate == nil {
+			return nil
+		}
+		if msg.CommissionRate.LT(sdk.NewDecWithPrec(5, 2)) {
+			return sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "commission can not be lower than 5%")
+		}
+	default:
+		return nil
+	}
+	return nil
```

Comment:
Is this check also executed for genesis validators?

Reply:
yes but you will have to check all validators set their gentx to at least 5%